src/client/datascience/baseJupyterSession.ts

```diff
@@ -104,38 +104,14 @@ export abstract class BaseJupyterSession implements IJupyterSession {
             traceInfo(`Unhandled message found: ${m.header.msg_type}`);
         };
     }
-    public dispose(): Promise<void> {
-        return this.shutdown();
+    public async dispose(): Promise<void> {
+        await this.shutdownImplementation(false);
     }
     // Abstracts for each Session type to implement
     public abstract waitForIdle(timeout: number): Promise<void>;
 
     public async shutdown(): Promise<void> {
-        this._isDisposed = true;
```

Comment:
As per Jupyter API shutdown should shutdown,
We aren't shutting things down and this method isn't used in our code, hence changing the code to shutdown the kernel when calling `shutdown` (including remote kernels).
This way 3rd party extensions using this API will get the desired behavior.

FYI - We call the dispose method to kill local kernels & to leave remote kernels running (i.e. we only ever call `dispose` hence our code isn't affected by this change)